### PR TITLE
CSHARP-4009: Replace dead links to api.mongodb.org with valid links.

### PR DESCRIPTION
**Pull request by rstam**
This PR is against master and is so that the links in future releases will be live.

Updating the gh-pages branch is a separate task.

### CODE REVIEW

File: Docs/landing/data/releases.toml [hunk 1]
```diff
@@ -64,22 +64,22 @@ current = "2.14.1"
 [[versions]]
   version = "2.2.4"
   docs = "./2.2/"
-  api = "http://api.mongodb.org/csharp/2.2"
+  api = "./2.2/apidocs"
```
Comment (rstam) [inline Docs/landing/data/releases.toml:67]: The updates in this file are the important ones.

File: Release Notes/Release Notes v1.7.md [hunk 2]
```diff
@@ -31,8 +31,7 @@ https://jira.mongodb.org/secure/IssueNavigator.jspa?mode=hide&requestId=12915
 
 Documentation on the C# driver can be found at:
 
-http://www.mongodb.org/display/DOCS/CSharp+Language+Center
-http://api.mongodb.org/csharp/current/
+http://mongodb.github.io/mongo-csharp-driver/
```
Comment (rstam) [inline Release Notes/Release Notes v1.7.md:34]: These old release notes are kind of moot, but I just wanted to get rid of all occurrences of `api.mongodb.org`.

Comment (JamesKovacs) [inline Release Notes/Release Notes v1.7.md:34]: Use `https` instead. Same for other release notes.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Minor changes requested.

File: Release Notes/Release Notes v1.7.md (hunk 2 above)
Comment (rstam) [reply Release Notes/Release Notes v1.7.md]: Done

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**